### ISSUE TITLE
If single test fails junit report writes no data

### ISSUE THREAD

**Issue by dhoulker** [webdriverio/webdriverio#631]
Hello, 

I've been playing around the new jenkins integration, especially the xunit reports.

I've been finding if any single test fails then my xunit report is zero bytes in size, and therefore it fails to report any of the other test results.

Is this a known issue, or is it something i might be doing wrong?

See below for the console output.

Thanks,

Dave

```
+ wdio wdio.conf.chrome.js
[31mError: expected promise to be fulfilled but it was rejected with [Error: element (h1=Careers) still without text after 8000ms][0m 
[1;30m    at screenshot() at /usr/local/lib/node_modules/webdriverio/lib/commands/saveScreenshot.js:30:17
    at saveScreenshot("./test/screenshots/ja02-careers_chrome.png") at /Users/Shared/Jenkins/Home/jobs/Acceptance tests/workspace/Functional Tests/Acceptance/test/commands/helpers.js:150:4[0m 
[31mError: expected promise to be fulfilled but it was rejected with [Error: element (h1=Unsubscribe from press releases) still without text after 8000ms][0m 
[1;30m    at elements("nav.secondary-nav") at /usr/local/lib/node_modules/webdriverio/lib/commands/isExisting.js:41:17
    at elements("nav.secondary-nav a", <Function>) at /Users/Shared/Jenkins/Home/jobs/Acceptance tests/workspace/Functional Tests/Acceptance/test/specs/desktop/04-newsResearcher/03-pressReleaseSubscribeForm.js:22:10[0m 
/usr/local/lib/node_modules/webdriverio/lib/reporter/xunit.js:314
            return (data.requestOptions.method || 'get').toUpperCase() + ' ' +
                                       ^
TypeError: Cannot read property 'method' of undefined
    at /usr/local/lib/node_modules/webdriverio/lib/reporter/xunit.js:314:40
    at XUnit.printLogTags (/usr/local/lib/node_modules/webdriverio/lib/reporter/xunit.js:217:20)
    at /usr/local/lib/node_modules/webdriverio/lib/reporter/xunit.js:313:14
    at Array.forEach (native)
    at XUnit.genTestsuite (/usr/local/lib/node_modules/webdriverio/lib/reporter/xunit.js:237:11)
    at /usr/local/lib/node_modules/webdriverio/lib/reporter/xunit.js:257:18
    at Array.forEach (native)
    at XUnit.genTestsuite (/usr/local/lib/node_modules/webdriverio/lib/reporter/xunit.js:237:11)
    at /usr/local/lib/node_modules/webdriverio/lib/reporter/xunit.js:257:18
    at Array.forEach (native)
+ true
```

**Comment by christian-bromann** [comment]
Nope that seems like a legit bug. Does this happen all the time?

**Comment by dhoulker** [comment]
Hi Chris,

Thanks for getting back to me so quickly - i think its a dodgy custom command that triggers it, but i've yet to verfiy that.

I have had it working with other failures, and reporting them correctly. Its started to happen since i've converted all my old tests from the old callback approach.

**Comment by christian-bromann** [comment]
@dhoulker seems that `waitForText` failed. Will have to check if this bug is especially caused by waitForX commands.

**Comment by christian-bromann** [comment]
Fixed. Will let you know here when it got released.

**Comment by dhoulker** [comment]
Great - thanks for picking this up quickly as usual!

**Comment by christian-bromann** [comment]
Fixed and published with `v3.2.0`
